core/org.wso2.carbon.core/src/test/java/org/wso2/carbon/core/util/KeyStoreManagerTest.java

```diff
@@ -0,0 +1,236 @@
+            KeyStore primaryKeystore = this.keyStoreManager.getPrimaryKeyStore();
+            assertSame(primaryKeystore, this.keyStore);
+        }
+    }
+
+    @Test(description = "Get trust store test")
+    public void testGetTrustStore() throws Exception {
+
+        try (MockedStatic<ServerConfiguration> serverConfiguration = mockStatic(ServerConfiguration.class);
+             MockedStatic<CryptoUtil>cryptoUtil = mockStatic(CryptoUtil.class);
+             MockedStatic<KeyStoreManager> keyStoreManager = mockStatic(KeyStoreManager.class);
+             MockedStatic<KeystoreUtils> keystoreUtils = mockStatic(KeystoreUtils.class)) {
+
+            cryptoUtil.when(CryptoUtil::lookupRegistryService).thenReturn(this.registryService);
+
+            keystoreUtils.when(() -> KeystoreUtils.getKeystoreInstance(anyString())).thenReturn(this.keyStore);
+
+            serverConfiguration.when(ServerConfiguration::getInstance).thenReturn(this.serverConfiguration);
+            when(this.serverConfiguration.getFirstProperty(
+                    RegistryResources.SecurityManagement.SERVER_TRUSTSTORE_FILE)).thenReturn(createPath(TRUSTSTORE_NAME).toString());
+            when(this.serverConfiguration.getFirstProperty(
+                    RegistryResources.SecurityManagement.SERVER_TRUSTSTORE_PASSWORD)).thenReturn(KEYSTORE_PASSWORD);
+            when(this.serverConfiguration.getFirstProperty(
+                    RegistryResources.SecurityManagement.SERVER_TRUSTSTORE_TYPE)).thenReturn(KEYSTORE_TYPE);
+
+            keyStoreManager.when(() -> KeyStoreManager.getInstance(anyInt())).thenReturn(this.keyStoreManager);
+
+            KeyStore trustStore = this.keyStoreManager.getTrustStore();
+            assertSame(trustStore, this.keyStore);
+        }
+    }
+
+    @Test(description = "Add KeyStore test")
+    public void testAddKeyStore() throws Exception {
+
+        byte[] keyStoreContent = readBytesFromFile(createPath(KEYSTORE_NAME).toString());
+
+        try (MockedStatic<CryptoUtil>cryptoUtil = mockStatic(CryptoUtil.class);
+             MockedStatic<KeyStoreManager> keyStoreManager = mockStatic(KeyStoreManager.class);
+             MockedStatic<KeystoreUtils> keystoreUtils = mockStatic(KeystoreUtils.class);
+             MockedStatic<KeyStoreUtil> keyStoreUtil = mockStatic(KeyStoreUtil.class)) {
+
+            cryptoUtil.when(CryptoUtil::lookupRegistryService).thenReturn(this.registryService);
+
+            keyStoreManager.when(() -> KeyStoreManager.getInstance(anyInt())).thenReturn(this.keyStoreManager);
+
+            keyStoreUtil.when(() -> KeyStoreUtil.isPrimaryStore(any())).thenReturn(false);
+            keyStoreUtil.when(() -> KeyStoreUtil.isTrustStore(any())).thenReturn(false);
+            keyStoreUtil.when(() -> KeyStoreUtil.getPrivateKeyAlias(any())).thenReturn("wso2carbon");
+
+            keystoreUtils.when(() -> KeystoreUtils.getKeystoreInstance(anyString())).thenReturn(this.keyStore);
+
+            // Mocking Registry interactions
```

Comment:
Shall we add the missing fullstop here. Let's check in other places as well.

Reply:
Fixed with https://github.com/wso2/carbon-kernel/pull/4091/commits/767822c4dacfe7deb0d9d42282a588835bdd690e